crates/gitbutler-core/src/git/commit_buffer.rs

```diff
@@ -0,0 +1,89 @@
+use anyhow::Result;
+use core::str;
+
+pub struct CommitBuffer {
+    heading: Vec<(String, String)>,
```

Comment:
A word of warning - enforcing UTF8 is going to fail occasionally when the encoding isn't UTF8 compatible. The 'safe' type for anything in Git `BString`, which can be edited and used similarly to `String`.

This mistake is already present throughout the codebase (sometimes with Paths, probably always with commit data), but I think it's important to try and avoid it.

`BString/&BStr` should be re-exported from `gix::` as well.